tests/cucumber/e2e/stepDefinitions/signup.Context.js

```diff
@@ -0,0 +1,31 @@
+const { Given, When, Then } = require ("@cucumber/cucumber")
+const { expect } = require("@playwright/test")
+const { SignUpPage } = require('../pageObjects/SignUpPage')
+const signUpPage = new SignUpPage
+
+Given('user has browsed into signup page', async function () {
+  await signUpPage.navigateSignUp()
+  
+  await expect(await signUpPage.navigateSignUp().getByText("SignUp")).toBeVisible()
+  });
```

Comment:
Why use the navigaetSignUp() function again to assert the expected outcome when you can simply use the selector to check if you redirect to the signup page.

```suggestion
Given('user has browsed into signup page', async function () {
  await signUpPage.navigateSignUp()
  
  await expect(await signUpPage.signupButtonSelector).toBeVisible()
  });
```